File: solutions/javascript/002-roman/solution.test.mjs
import test from "node:test";
import assert from "node:assert/strict";
import { toRoman, fromRoman } from "./solution.mjs";

test("toRoman / fromRoman", () => {
  assert.equal(toRoman(1), "I");
  assert.equal(toRoman(4), "IV");
  assert.equal(toRoman(9), "IX");
  assert.equal(toRoman(40), "XL");
  assert.equal(toRoman(44), "XLIV");
  assert.equal(toRoman(1994), "MCMXCIV");
  assert.equal(toRoman(3999), "MMMCMXCIX");
  assert.equal(fromRoman("MCMXCIV"), 1994);
  assert.equal(fromRoman("lviii"), 58);
  assert.throws(() => toRoman(0));
  assert.throws(() => fromRoman("ABC"));
  for (const n of [1, 4, 9, 40, 44, 1994, 3999]) {
    assert.equal(fromRoman(toRoman(n)), n);
  }
});

File: solutions/python/002-roman/solution_test.py
from solution import to_roman, from_roman

assert to_roman(1) == "I"
assert to_roman(4) == "IV"
assert to_roman(9) == "IX"
assert to_roman(40) == "XL"
assert to_roman(44) == "XLIV"
assert to_roman(1994) == "MCMXCIV"
assert to_roman(3999) == "MMMCMXCIX"
assert from_roman("MCMXCIV") == 1994
assert from_roman("lviii") == 58
try:
    to_roman(0)
    raise AssertionError("expected fail")
except ValueError:
    pass
try:
    from_roman("ABC")
    raise AssertionError("expected fail")
except ValueError:
    pass
for n in [1, 4, 9, 40, 44, 1994, 3999]:
    assert from_roman(to_roman(n)) == n
print("ok")
